doc/semieunit.xml

```diff
@@ -0,0 +1,292 @@
+##
+#############################################################################
+##
+
+<#GAPDoc Label="IsMcAlisterTripleSemigroup">
+  <ManSection>
+    <Prop Name = "IsMcAlisterTripleSemigroup" Arg = "S"/>
+    <Returns><K>true</K> or <K>false</K>.</Returns>
+    <Description>
+      A <E>McAlister triple semigroup</E> is an E-unitary inverse semigroup
+      <Ref Oper= "IsEUnitaryInverseSemigroup" BookName= "Semigroups" />
+      defined by a partial order, a semilattice and a group which acts by
+      order automorphisms on the partial order. All E-unitary inverse
+      semigroups can be represented as a McAlister triple semigroup and all
+      McAlister triple semigroups define E-unitary inverse semigroups. <P/>
+
+      We use two <Ref Oper= "Digraph" BookName= "Digraphs" /> objects to
+      represent the partial order and the semilattice of a McAlister triple.
+      These will satisfy <Ref Oper= "IsPartialOrderDigraph" 
+      BookName= "Digraphs" /> and <Ref Oper= "IsJoinSemilatticeDigraph"
+      BookName= "Digraphs" />, respecitvely. <P/>
+
+      A rigorous treatment of McAlister triples may be found in Chapter 5
+      of <Cite Key = "Howie1995aa"/>. The precise conditions
+      for a triple in &GAP; are as follows:
+
+      <List>
+        <Mark> M1 </Mark>
+        <Item>
+          <C>Y</C> is a semilattice which is a subdigraph of <C>X</C>.
+        </Item>
+        <Mark> M2 </Mark> 
+        <Item>
+          For all <C>A</C> in <C>X</C> and for all <C>B</C> in <C>Y</C>:
+          if <C>A</C> <M>\leq</M> <C>B</C> then <C>A</C> in <C>Y</C>.
+        </Item>
+        <Mark> M3 </Mark> 
+        <Item>
+          For every <C>A</C> in <C>X</C> there exists some <C>B</C> in
+          <C>Y</C> and <C>g</C> in <C>G</C> such that <C>A</C> is equal to
+          <C>B</C> acted on by <C>g</C>.
+        </Item>
+        <Mark> M4 </Mark> 
+        <Item>
+          For all <C>g</C> in <C>G</C> we have that the intersection of <C>Y</C>
+          acted on by <C>g</C> with <C>Y</C> is non-empty.
+        </Item>
+      </List>
+
+     <Example><![CDATA[
+gap> x := Digraph([[1], [1, 2], [1, 2, 3], [1, 4], [1, 4, 5]]);
+<digraph with 5 vertices, 11 edges>
+gap> y := InducedSubdigraph(x, [1, 2, 3]);
+<digraph with 3 vertices, 6 edges>
+gap> A := AutomorphismGroup(x);
+Group([ (2,4)(3,5) ])
+gap> S := McAlisterTripleSemigroup(A, OnPoints, x, y);
+<McAlister triple semigroup over Group([ (2,4)(3,5) ])>
+gap> T := McAlisterTripleSemigroup(A, x, y);
+<McAlister triple semigroup over Group([ (2,4)(3,5) ])>
+gap> T = S;
+true
+gap> GeneratorsOfSemigroup(T);
+[ (1, ()), (2, ()), (3, ()), (1, (2,4)(3,5)) ]
+gap> AsSemigroup(IsPartialPermSemigroup, T);
+<inverse partial perm monoid of size 4, rank 4 with 4 generators>]]></Example>
+    </Description>
+  </ManSection>
+<#/GAPDoc>
+
+<#GAPDoc Label="McAlisterTripleSemigroup">
+  <ManSection>
+    <Oper Name = "McAlisterTripleSemigroup" Arg = "G, act, X, Y"
```

Comment:
I think the order of arguments in the 4-argument versions should be the same as in the 3-argument versions.

So you have `G, X, Y, act` or `G, X, Y`, and similarly `G, X, sub_ver, act` and `G, X, sub_ver`.